pkg/platform/runtime/buildscript/buildscript.go

```diff
@@ -91,6 +91,11 @@ func indent(s string) string {
 func (s *Script) String() string {
 	buf := strings.Builder{}
 	for _, assignment := range s.Expr.Let.Assignments {
+		if assignment.Name == "requirements" {
+			buf.WriteString(requirementsString(assignment))
+			buf.WriteString("\n")
+			continue
+		}
 		buf.WriteString(assignmentString(assignment))
 		buf.WriteString("\n")
```

Comment:
What would make me over-the-moon-happy would be to have something like this:

```
if assignment.Name == "requirements" {
  assignment = &Assignment{assignment.Name, transformRequirements(assignment.Value)}
}
buf.WriteString(assignmentString(assignment))
buf.WriteString("\n")
```

where `transformRequirements` iterates over the given `buildexpression.Value` (whose `List` field is non-nil) and returns a new `buildexpression.Value` whose `List` field is constructed of `buildexpression.Ap` values of the name "Req" and whose arguments are `buildexpression.Value`s with `Assignment` fields of the appropriate "name" and "version" buildscript fields. (Note: you will need to add a method to https://github.com/ActiveState/cli/blame/master/pkg/platform/model/version_constraints.go that converts a `[]bpModel.VersionRequirement` to a string using the existing `versionConstraintsToString()` and use that returned value for the "version" field; your existing code does not handle complex version constraints, while the previous buildscript format did.)

By doing so, we should be able to avoid the entire `requirementsString()` function and piggy-back off of the existing `valueString()` and `apString()` functions. (Note: you may need to modify `apString()` to not print newlines if the name is "Req"/constant.)

Reply:
Was able to get this to work. Thanks for the pointers. It's not the _exact_ syntax you provided but it's close.